### ISSUE TITLE
Transformers should support tensors

### ISSUE THREAD

**Issue by peastman** [deepchem/deepchem#1170]
Transformers are currently implemented to operate on numpy arrays.  We should consider extending them so they can also operate on tensors.  This would allow them to be used in Tensorflow's eager mode.  In #915 we discussed letting `predict_on_batch()` accept tensors, but right now the `transformers` argument wouldn't work.  This also would let MultiTaskFitTransformRegressor support `make_estimator()`.

**Comment by lilleswing** [comment]
+1 ANI1 Transformer is a good example. (Although our latest benchmarks will probably move them back over to cpu-land for more performance)

**Comment by rbharath** [comment]
+1. This sounds like a useful addition.

**Comment by rbharath** [comment]
Do we still think expanding transformers to work on tensors would be a useful features? Off the cuff, it seems we could just implement the transformation within a layer rather than expanding transformers to work on tensors but I might be missing the full picture

CC @peastman

**Comment by peastman** [comment]
> it seems we could just implement the transformation within a layer rather than expanding transformers

That would be more or less equivalent to this.  One way or another, we want to be able to put tensors into the transformation code and then backpropagate gradients through it.  The advantage of transformers is that you can plug them into existing models.  You don't need to modify the model code.
